packages/emotion-primitives/src/css.js

```diff
@@ -3,71 +3,82 @@ import transform from 'css-to-react-native'
-  } else {
-    vals = interleave(args)
-  }
+function handleInterpolation(interpolation: *, i: number, arr: Array<*>) {
+  let type = typeof interpolation
 
-  let styles = []
-  let buffer = ''
-  let lastType
+  if (type === 'function') {
+    if (this === undefined && process.env.NODE_ENV !== 'production') {
+      console.error(
+        'Interpolating functions in css calls is not allowed.\n' +
+          'If you want to have a css call based on props, create a function that returns a css call like this\n' +
+          'let dynamicStyle = (props) => css`color: ${props.color}`\n' +
+          'It can be called directly with props or interpolated in a styled call like this\n' +
+          'let SomeComponent = styled.View`${dynamicStyle}`'
+      )
+    } else {
+      handleInterpolation.call(
+        this,
+        interpolation(
+          // $FlowFixMe
+          this.mergedProps
+        ),
+        i,
+        arr
+      )
+    }
+    return
+  }
+  let isIrrelevant = interpolation == null || type === 'boolean'
+  let isRnStyle =
+    (type === 'object' && !Array.isArray(interpolation)) || type === 'number'
+  if (lastType === 'string' && (isRnStyle || isIrrelevant)) {
+    let converted = convertStyles(buffer)
+    if (converted !== undefined) {
+      styles.push(converted)
+    }
+    buffer = ''
+  }
+  if (isIrrelevant) {
+    return
+  }
 
-  let handleInterpolation = (interpolation: *, i: number, arr: Array<*>) => {
-    let type = typeof interpolation
+  if (type === 'string') {
+    buffer += interpolation
 
-    if (type === 'function') {
-      if (this === undefined && process.env.NODE_ENV !== 'production') {
-        console.error(
-          'Interpolating functions in css calls is not allowed.\n' +
-            'If you want to have a css call based on props, create a function that returns a css call like this\n' +
-            'let dynamicStyle = (props) => css`color: ${props.color}`\n' +
-            'It can be called directly with props or interpolated in a styled call like this\n' +
-            'let SomeComponent = styled.View`${dynamicStyle}`'
-        )
-      } else {
-        handleInterpolation(interpolation(this.mergedProps), i, arr)
-      }
-      return
-    }
-    let isFalsy = interpolation == null || type === 'boolean'
-    let isRnStyle =
-      (type === 'object' && !Array.isArray(interpolation)) || type === 'number'
-    if (lastType === 'string' && (isRnStyle || isFalsy)) {
+    if (arr.length - 1 === i) {
       let converted = convertStyles(buffer)
       if (converted !== undefined) {
         styles.push(converted)
       }
       buffer = ''
     }
-    if (isFalsy) {
-      return
-    }
+  }
+  if (isRnStyle) {
+    styles.push(interpolation)
+  }
+  if (Array.isArray(interpolation)) {
+    interpolation.forEach(handleInterpolation, this)
+  }
+  lastType = type
+}
 
-    if (type === 'string') {
-      buffer += interpolation
+export function css(...args: any) {
+  let vals
+  styles = []
```

Comment:
I'm not sure if it's really worth it/will make a difference. We create a bunch of arrays since `css-to-react-native` accepts styles like this `[['color', 'hotpink'], ['background-color', 'yellow']]` so I don't thinking creating one less here will matter. Also, I don't think mutating it is a good idea since `StyleSheet.flatten` could have memoization/could have it in the future and using the same array would break that.(that might never happen but I think passing an object to a function that we don't control and mutating it isn't a great idea)